services/bonita-time-tracker/src/main/java/org/bonitasoft/engine/tracking/TimeTracker.java

```diff
@@ -28,151 +31,332 @@
+        return new FlushThread(this);
+    }
+
+
+    private void internalStartTracking() {
+        if (startTracking) {
+            if (logger.isLoggable(getClass(), TechnicalLogSeverity.WARNING)) {
+                this.logger.log(getClass(), TechnicalLogSeverity.WARNING,
+                        "Starting Time tracker tracking...");
+            }
+            if (!flushThread.isStarted()) {
+                flushThread.start();
+            }
+            for (final FlushEventListener listener : getActiveFlushEventListeners()) {
+                listener.notifyStartTracking();
+            }
+            if (logger.isLoggable(getClass(), TechnicalLogSeverity.WARNING)) {
+                this.logger.log(getClass(), TechnicalLogSeverity.WARNING,
+                        "Time tracker tracking is activated. This may not be used in production as performances may be strongly impacted.");
+            }
+        }
+    }
+
+    private void internalStopTracking() {
+        if (isTracking()) {
+            if (logger.isLoggable(getClass(), TechnicalLogSeverity.WARNING)) {
+                this.logger.log(getClass(), TechnicalLogSeverity.WARNING,
+                        "Stopping Time tracker tracking...");
+            }
+            if (flushThread.isStarted()) {
+                flushThread.interrupt();
+            }
+            for (final FlushEventListener listener : getActiveFlushEventListeners()) {
+                listener.notifyStopTracking();
+            }
+            if (logger.isLoggable(getClass(), TechnicalLogSeverity.WARNING)) {
+                this.logger.log(getClass(), TechnicalLogSeverity.WARNING,
+                        "Time tracker tracking is deactivated.");
+            }
         }
-        flushThread = createFlushThread(logger, clock, flushIntervalInSeconds);
+    }
+
+    public boolean isTracking() {
+        return flushThread.isStarted();
+    }
 
+    public long getFlushIntervalInMS() {
+        return flushIntervalInMS;
     }
 
-    FlushThread createFlushThread(TechnicalLoggerService logger, Clock clock, int flushIntervalInSeconds) {
-        return new FlushThread(clock, flushIntervalInSeconds, this, logger);
+    public void setFlushIntervalInSeconds(long flushIntervalInSeconds) {
+        this.flushIntervalInMS = flushIntervalInSeconds * 1000;
     }
 
-    public boolean isTrackable(final String recordName) {
-        return enabled && started && activatedRecords.contains(recordName);
+    public void setFlushIntervalInMS(long flushIntervalInMS) {
+        this.flushIntervalInMS = flushIntervalInMS;
     }
 
-    public void track(final String recordName, final String recordDescription, final long duration) {
+    public Clock getClock() {
+        return clock;
+    }
+
+    public String getStatus() {
+        final StringBuilder sb = new StringBuilder();
+        sb.append("-----");
+        sb.append("\n");
+
+        sb.append("Time Tracker '");
+        sb.append(this.getClass().getName());
```

Comment:
look at the final "full" implementation, there are many lines, seems not a good idea to chain all of them. In addition to that my IDE is recommending not to chain to avoid a new StringBuilder allocation...